### DISCUSSION TITLE
What is the release concept of CAPO?

### DISCUSSION THREAD

**Discussion by cwrau** [kubernetes-sigs/cluster-api-provider-openstack#1676] [Q&A] [answered]
Somehow there are releases, like https://github.com/kubernetes-sigs/cluster-api-provider-openstack/releases/tag/v0.8.0, which don't include previously merged PRs, like https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1648.

Is this a mistake? Or what is the concept here?

**Comment by mdbooth** [comment]
It looks like #1648 is included in v0.8.0, or am I missing something?

**Reply by cwrau** [reply]
Is it? It's at least not in the changelog

**Reply by cwrau** [reply]
Ah, I can see it in https://github.com/kubernetes-sigs/cluster-api-provider-openstack/compare/v0.7.3...v0.8.0

So it's just a bug in the changelog generation?

**Reply by mdbooth** [reply]
It was more of a choice in changelog generation, tbh: I decided not to repeat the changelogs between pre-releases. However, it sounds like I should reconsider that.

**Reply by cwrau** [reply]
That would be great, I'd assume most people, like me, just watch for full releases and don't look at the prereleases

**Reply by wwentland** [reply]
I agree with @cwrau here and would personally prefer the following behaviour:

- All changes between proper (e.g v0.7.3..v0.8.0) releases are included in release notes for releases
- Incremental changes between pre-releases are included in their respective release notes

This would result in some repetition, but as an end user I am primarily interested in changes between versions I am genuinely using and those are by and large only major ones.

A nice side-effect is that contributors see their contribution included in a "real" major release, rather than mostly pre-releases.

**Comment by mdbooth** [comment, marked as answer]
To summarise the above: the issue was that although was fixed in the release, it was only mentioned in a pre-release release note rather than the actual release notes.

For future releases:
* Pre-releases will contain a changelog from the previous pre-release, or the last stable release if there isn't one.
* Full releases will contain a changelog from the last stable release.

We should document this decision in https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/RELEASE.md so the release monkey doesn't forget! Any volunteer?

**Comment by wwentland** [comment]
> We should document this decision in https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/RELEASE.md so the release monkey doesn't forget! Any volunteer?

I'd be happy to take a look at this in the coming days.
